examples/js/utils/SceneUtils.js

```diff
@@ -37,6 +37,52 @@
 
 	}
 
+	//Convert multi-material mesh into group of mesh with single material
+	function createMeshesFromMultiMaterialMesh( mesh ) {
+
+		const group = new THREE.Group();
+
+		//split groups by material index
+		const materials = mesh.material;
+
+		if ( ! ( materials instanceof Array ) ) {
+
+			console.error( 'Not a multimaterial mesh.' );
+			return;
+
+		}
+
+		materials.forEach( ( mat, index ) => {
+
+			const singleIndexGroups = mesh.geometry.groups.filter( g => g.materialIndex == index );
+			const singleIndices = mesh.geometry.index.array.filter( ( val, i ) => {
+
+				let belong = false;
+				singleIndexGroups.forEach( g => {
+
+					belong |= ( g.start <= i && g.count + g.start > i );
+
+				} );
+
+				return belong;
+
+			} );
+			const singleGeo = mesh.geometry.clone();
+			singleGeo.setIndex( new THREE.BufferAttribute( new Uint16Array( singleIndices ), 1 ) );
```

Comment:
Just do:
```js
singleGeo.setIndex( singleIndices );
```
In this way, `setIndex()` picks the appropriate buffer attribute type for you.